server/map_rpc_server.go

```diff
@@ -350,6 +324,12 @@ func (t *TrillianMapServer) SetLeaves(ctx context.Context, req *trillian.SetMapL
 	if err != nil {
 		return nil, err
 	}
+	indexValidator := newRequestIndexValidator(hasher.Size())
+	for _, l := range req.Leaves {
```

Comment:
Consider adding this loop's index to error messages instead of the byte array: `for i, l := range ...` + `fmt.Errorf("key #%d: ...", i, ...)`
Advantage: If an RPC client sends a bonkersly long leaf index, we won't send it back to them.

Reply:
Done.